### DISCUSSION TITLE
Pull docker image from GitHub Packages - GITHUB_TOKEN doesn't work, while my personal token works

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#28757] [Actions] [answered]
I'm trying to pull docker image, using standard method:

`echo ${{ secrets.GITHUB_TOKEN }} |docker login ghcr.io -u ${{ github.actor }} --password-stdin
            docker pull ghcr.io/myorg/myimage:latest`
            
  Using secrets.GITHUB_TOKEN - I'm getting "Login Succeeded", but "docker pull" fails with "Error response from daemon: denied".
  But if I'm using my personal token,  "docker pull" works OK.
  
How to debug/fix that?

Thanks, Vitaly

**Comment by danielmeppiel** [comment, marked as answer]
You should check if the default `GITHUB_TOKEN` permissions are set to `restrictive` or `permissive` as documented [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token).

You can quickly check your action's run log, _Set up job_ section, and check what level of permissions the `GITHUB_TOKEN` had. You won't be able to pull images if that's set to `restrictive`.

Let me know if that helps!

**Reply by vitalyk-multinarity** [reply]
Daniel, 
Sorry, I didn't mention that in my question, but I already checked that - I see "Packages: write" under my job log, "GITHUB_TOKEN Permissions".
Thank you, Vitaly

**Reply by danielmeppiel** [reply]
Did you give the repository where the workflow lives explicit access to the image you are trying to pull?

See [this article](https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio)

**Reply by vitalyk-multinarity** [reply]
No, I didn't. Many thanks - I guess this is the issue!
